File: learning/src/main/scala/com/leetcode/learning/recursion/TreeNode.scala
package com.leetcode.learning.recursion

class TreeNode(_value: Int = 0, _left: TreeNode = null, _right: TreeNode = null) {
  val value: Int = _value
  var left: TreeNode = _left
  var right: TreeNode = _right

  override def toString: String = s"${this.value}"
}

object TreeNode {
  def apply(v: Int): TreeNode = new TreeNode(v)

  def apply(v: Int, l: TreeNode): TreeNode = new TreeNode(v, l)

  def apply(v: Int, l: TreeNode, r: TreeNode): TreeNode = new TreeNode(v, l, r)
}

File: learning/src/main/java/com/leetcode/learning/bst/TreeNode.java
package com.leetcode.learning.bst;

final class TreeNode {
    final int val;
    TreeNode left, right;

    TreeNode(int val) {
        this.val = val;
    }

    TreeNode(int val, TreeNode left, TreeNode right) {
        this.val = val;
        this.left = left;
        this.right = right;
    }

    @Override
    public String toString() {
        return String.format("Node( %d )", val);
    }
}
